dot/state/block.go

```diff
@@ -250,14 +250,62 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetAllDescendants gets all the descendants even if hash is not stored in memory.
+func (bs *BlockState) GetAllDescendants(hash common.Hash) ([]common.Hash, error) {
+	allDescendants, err := bs.bt.GetAllDescendants(hash)
+	if err != nil && !errors.Is(err, blocktree.ErrNodeNotFound) {
+		return nil, err
+	}
+
+	if err == nil {
+		return allDescendants, nil
+	}
+
+	header, err := bs.GetHeader(hash)
+	if err != nil {
+		return nil, fmt.Errorf("getting header from hash %s: %w", hash, err)
```

Comment:
hash information would be added by `GetHeader` and shouldn't be duplicated here

```suggestion
		return nil, fmt.Errorf("getting header: %w", err)
```